examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -0,0 +1,530 @@
+        self.g = g
+        self.embed_size = embed_size
+
+        # create weight embeddings for each node for each relation
+        self.embeds = nn.ParameterDict()
+        for ntype in g.ntypes:
+            if ntype in exclude:
+                continue
+            embed = nn.Parameter(th.Tensor(g.number_of_nodes(ntype), self.embed_size))
+            self.embeds[ntype] = embed
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        for emb in self.embeds.values():
+            nn.init.xavier_uniform_(emb)
+
+    def forward(self, block=None):
+        return self.embeds
+
+
+class RelGraphConvLayer(nn.Module):
+    r"""Relational graph convolution layer.
+
+    Parameters
+    ----------
+    in_feat : int
+        Input feature size.
+    out_feat : int
+        Output feature size.
+    ntypes : list[str]
+        Node type names
+    rel_names : list[str]
+        Relation names.
+    weight : bool, optional
+        True if a linear layer is applied after message passing. Default: True
+    bias : bool, optional
+        True if bias is added. Default: True
+    activation : callable, optional
+        Activation function. Default: None
+    self_loop : bool, optional
+        True to include self loop message. Default: False
+    dropout : float, optional
+        Dropout rate. Default: 0.0
+    """
+    def __init__(self,
+                 in_feat,
+                 out_feat,
+                 ntypes,
+                 rel_names,
+                 *,
+                 weight=True,
+                 bias=True,
+                 activation=None,
+                 self_loop=False,
+                 dropout=0.0):
+        super(RelGraphConvLayer, self).__init__()
+        self.in_feat = in_feat
+        self.out_feat = out_feat
+        self.ntypes = ntypes
+        self.rel_names = rel_names
+        self.bias = bias
+        self.activation = activation
+        self.self_loop = self_loop
+
+        self.conv = dglnn.HeteroGraphConv({
+                rel : dglnn.GraphConv(in_feat, out_feat, norm='right', weight=False, bias=False)
+                for rel in rel_names
+            })
+
+        self.use_weight = weight
+        if self.use_weight:
+            self.weight = nn.ModuleDict({
+                rel_name: nn.Linear(in_feat, out_feat, bias=False)
+                for rel_name in self.rel_names
+            })
+
+        # weight for self loop
+        if self.self_loop:
+            self.loop_weights = nn.ModuleDict({
+                ntype: nn.Linear(in_feat, out_feat, bias=bias)
+                for ntype in self.ntypes
+            })
+
+        self.dropout = nn.Dropout(dropout)
+
+        self.reset_parameters()
+
+    
+    def reset_parameters(self):
+        if self.use_weight:
+            for layer in self.weight.values():
+                layer.reset_parameters()
+
+        if self.self_loop:
+            for layer in self.loop_weights.values():
+                layer.reset_parameters()
```

Comment:
Strictly speaking, this should also reset the parameters of `self.conv` although currently `HeteroGraphConv` lacks a function `reset_parameters`. Could you try adding a function for it? You can loop over `HeteroGraphConv.mods` and reset the parameters of a module if it has function `reset_parameters`.

Reply:
I think this is a great idea, but should be addressed in a separate PR.  I don't know if we'll want to make similar considerations for MXNet...etc so that behavior of passing in external weights is consistent.  This change would also require more extensive testing I think.